### ISSUE TITLE
API Doc Improvements - Tracking

### ISSUE THREAD

**Issue by domdivakaruni** [dmlc/mxnet#6000]
## Let's fix our docs!
The effort to spruce up MXNet's Python API docs is picking up steam but could use **your** help to add that extra oomph! Many of the APIs now have proper explanations and good examples, but we still have much more to go!

![alt text](https://s3.amazonaws.com/mxnetdoc/fist.jpg "Make our docs great again!")

## Here’s how you can join in and help:

1. ### **Pick a few functions (or at least 1 :smile: ) from the list below**
    - Ensure you only pick ones that don’t have a current owner assigned and you can submit within 2 business days. Post a comment to the issue with the ones you’ve selected.

1. ### **Look up current documentation**
    - Find whats currently documented for the function at the [mxnet.io python API docs](http://mxnet.io/api/python/index.html) or within the C++ code in git.

1. ### **Write up:**
      -	A clear and concise description of the function and its behavior. 
      -	List and describe each parameter with the valid input values, whether it is required or optional, and default value if the parameter is optional.
      -	Illustrate the function and parameter behavior using examples
      -	Refer to these examples for guidance:- [Embedding](http://mxnet.io/api/python/ndarray.html#mxnet.ndarray.Embedding) , [ROIPooling]( http://mxnet.io/api/python/ndarray.html#mxnet.ndarray.ROIPooling) , [Reshape]( http://mxnet.io/api/python/ndarray.html#mxnet.ndarray.Reshape)

1. ### **Submit your work**
      -	Creating a new issue with [API docs submission] ‘your function name(s)’ in the title and paste your write-up markdown
               **OR**
      -	Creating a pull request for a change to the appropriate .cc or .py file. Instructions here: [How to make a pull request for docs]( https://s3.amazonaws.com/mxnetdoc/API+Doc+Contributions%281%29.pdf)

## API List
| #   | Function Name                                                      | Package       | Contributor          | Status           | 
|-----|--------------------------------------------------------------------|---------------|----------------------|------------------| 
| 1   | <ul><li> - [ ] IRHeader   </li></ul>                               | recordio      |                      |                  | 
| 2   | <ul><li> - [ ] pack    </li></ul>                                  | recordio      |                      |                  | 
| 3   | <ul><li> - [ ] pack_img    </li></ul>                              | recordio      |                      |                  | 
| 4   | <ul><li> - [ ] unpack    </li></ul>                                | recordio      |                      |                  | 
| 5   | <ul><li> - [ ] unpack_img    </li></ul>                            | recordio      |                      |                  | 
| 6   | <ul><li> - [ ] BilinearSampler    </li></ul>                       | ndarray       |  @Lyken17 |                  | 
| 7   | <ul><li> - [ ] Correlation    </li></ul>                           | ndarray       | @Lyken17 |                  | 
| 8   | <ul><li> - [ ] GridGenerator    </li></ul>                         | ndarray       |   @Lyken17 |                  | 
| 9   | <ul><li> - [ ] IdentityAttachKLSparseReg    </li></ul>             | ndarray       | @Lyken17 |                  | 
| 10  | <ul><li> - [ ] SpatialTransformer    </li></ul>                    | ndarray       | @Lyken17 |                  | 
| 11  | <ul><li> - [ ] UpSampling    </li></ul>                            | ndarray       | @Lyken17 |                  | 
| 12  | <ul><li> - [ ] adam_update    </li></ul>                           | ndarray       |                      |                  | 
| 13  | <ul><li> - [ ] ones_like    </li></ul>                             | ndarray       | @yuruofeifei |                  | 
| 14  | <ul><li> - [ ] random_exponential    </li></ul>                    | ndarray       | @asmushetzel |                  | 
| 15  | <ul><li> - [ ] random_gamma    </li></ul>                          | ndarray       | @asmushetzel |                  | 
| 16  | <ul><li> - [ ] random_generalized_negative_binomial    </li></ul>  | ndarray       |  @asmushetzel  |                  |
| 17  | <ul><li> - [ ] random_negative_binomial    </li></ul>              | ndarray       | @asmushetzel |                  | 
| 18  | <ul><li> - [ ] random_poisson    </li></ul>                        | ndarray       | @asmushetzel |                  | 
| 19  | <ul><li> - [x] rmsprop_update    </li></ul>                        | ndarray       | @jiajiechen  |                  | 
| 20  | <ul><li> - [x] rmspropalex_update    </li></ul>                    | ndarray       |  @jiajiechen   |                  | 
| 21  | <ul><li> - [ ] sample_exponential    </li></ul>                    | ndarray       |  @asmushetzel                    |                  | 
| 22  | <ul><li> - [ ] sample_gamma    </li></ul>                          | ndarray       |  @asmushetzel                    |                  | 
| 23  | <ul><li> - [ ] sample_generalized_negative_binomial    </li></ul>  | ndarray       |    @asmushetzel                |                  | 
| 24  | <ul><li> - [ ] sample_negative_binomial    </li></ul>              | ndarray       | @asmushetzel                   |                  | 
| 25  | <ul><li> - [ ] sample_normal    </li></ul>                         | ndarray       | @asmushetzel |                  | 
| 26  | <ul><li> - [ ] sample_poisson    </li></ul>                        | ndarray       |   @asmushetzel           |                  | 
| 27  | <ul><li> - [ ] sample_uniform    </li></ul>                        | ndarray       |  @asmushetzel            |                  | 
| 28  | <ul><li> - [ ] sgd_mom_update    </li></ul>                        | ndarray       |                      |                  | 
| 29  | <ul><li> - [ ] sgd_update    </li></ul>                            | ndarray       |                      |                  | 
| 30  | <ul><li> - [ ] smooth_l1    </li></ul>                             | ndarray       |                      |                  | 
| 31  | <ul><li> - [ ] softmax_cross_entropy    </li></ul>                 | ndarray       | @yuruofeifei |                  | 
| 32  | <ul><li> - [ ] waitall    </li></ul>                               | ndarray       |                      |                  | 
| 33  | <ul><li> - [ ] zeros_like    </li></ul>                            | ndarray       | @yuruofeifei |                  | 
| 34  | <ul><li> - [ ] Caffe    </li></ul>                                 | metric        |                      |                  | 
| 35  | <ul><li> - [ ] Torch    </li></ul>                                 | metric        |                      |                  | 
| 36  | <ul><li> - [ ] check_label_shapes    </li></ul>                    | metric        |                      |                  | 
| 37  | <ul><li> - [ ] Bilinear    </li></ul>                              | initializer   |                      |                  | 
| 38  | <ul><li> - [ ] FusedRNN    </li></ul>                              | initializer   |                      |                  | 
| 39  | <ul><li> - [ ] LSTMBias    </li></ul>                              | initializer   |                      |                  | 
| 40  | <ul><li> - [ ] LogValidationMetricsCallback    </li></ul>          | callback      |                      |                  | 
| 41  | <ul><li> - [ ] _init_torch_module    </li></ul>                    | torch         |                      |                  | 
| 42  | <ul><li> - [ ] _make_torch_function    </li></ul>                  | torch         |                      |                  | 
| 43  | <ul><li> - [ ] CustomOp.assign    </li></ul>                       | operator      |                      |                  | 
| 44  | <ul><li> - [ ] NameManager    </li></ul>                           | name          |                      |                  | 
| 45  | <ul><li> - [ ] Monitor    </li></ul>                               | monitor       |                      |                  | 
| 46  | <ul><li> - [ ] CastAug    </li></ul>                               | image         |                      |                  | 
| 47  | <ul><li> - [ ] CenterCropAug    </li></ul>                         | image         |                      |                  | 
| 48  | <ul><li> - [ ] ColorJitterAug    </li></ul>                        | image         |                      |                  | 
| 49  | <ul><li> - [ ] ColorNormalizeAug    </li></ul>                     | image         |                      |                  | 
| 50  | <ul><li> - [ ] CreateAugmenter    </li></ul>                       | image         |                      |                  | 
| 51  | <ul><li> - [ ] HorizontalFlipAug    </li></ul>                     | image         |                      |                  | 
| 52  | <ul><li> - [ ] LightingAug    </li></ul>                           | image         |                      |                  | 
| 53  | <ul><li> - [ ] RandomCropAug    </li></ul>                         | image         |                      |                  | 
| 54  | <ul><li> - [ ] RandomOrderAug    </li></ul>                        | image         |                      |                  | 
| 55  | <ul><li> - [ ] RandomSizedCropAug    </li></ul>                    | image         |                      |                  | 
| 56  | <ul><li> - [ ] ResizeAug    </li></ul>                             | image         |                      |                  | 
| 57  | <ul><li> - [ ] center_crop    </li></ul>                           | image         |                      |                  | 
| 58  | <ul><li> - [ ] color_normalize    </li></ul>                       | image         |                      |                  | 
| 59  | <ul><li> - [ ] fixed_crop    </li></ul>                            | image         |                      |                  | 
| 60  | <ul><li> - [ ] imdecode    </li></ul>                              | image         |                      |                  | 
| 61  | <ul><li> - [ ] random_crop    </li></ul>                           | image         |                      |                  | 
| 62  | <ul><li> - [ ] random_size_crop    </li></ul>                      | image         |                      |                  | 
| 63  | <ul><li> - [ ] resize_short    </li></ul>                          | image         |  @jhwei  |                  | 
| 64  | <ul><li> - [ ] scale_down    </li></ul>                            | image         | @jhwei   |                  | 
| 65  | <ul><li> - [ ] ImageIter.augmentation_transform    </li></ul>      | image         |                      |                  | 
| 66  | <ul><li> - [ ] ImageIter.imdecode    </li></ul>                    | image         |                      |                  | 
| 67  | <ul><li> - [ ] ImageIter.postprocess_data    </li></ul>            | image         |                      |                  | 
| 68  | <ul><li> - [ ] ImageIter.read_image    </li></ul>                  | image         |                      |                  | 
| 69  | <ul><li> - [ ] ImageIter.check_data_shape    </li></ul>            | image         |                      |                  | 
| 70  | <ul><li> - [ ] ImageIter.check_valid_image    </li></ul>           | image         |                      |                  | 
| 71  | <ul><li> - [ ] Executor._get_dict    </li></ul>                    | executor      |                      |                  | 
| 72  | <ul><li> - [ ] _monitor_callback_wrapper    </li></ul>             | executor      |                      |                  | 
| 73  | <ul><li> - [ ] MultiBoxDetection    </li></ul>                     | contrib       |                      |                  | 
| 74  | <ul><li> - [ ] MultiBoxPrior    </li></ul>                         | contrib       |                      |                  | 
| 75  | <ul><li> - [ ] MultiBoxTarget    </li></ul>                        | contrib       |                      |                  | 
| 76  | <ul><li> - [ ] Proposal    </li></ul>                              | contrib       |                      |                  | 
| 77  | <ul><li> - [ ] count_sketch    </li></ul>                          | contrib       |                      |                  | 
| 78  | <ul><li> - [ ] fft    </li></ul>                                   | contrib       |                      |                  | 
| 79  | <ul><li> - [ ] ifft    </li></ul>                                  | contrib       |                      |                  | 
| 80  | <ul><li> - [ ] set_is_training    </li></ul>                       | contrib       |                      |                  | 
| 81  | <ul><li> - [ ] test    </li></ul>                                  | contrib       |                      |                  | 
| 82  | <ul><li> - [ ] train    </li></ul>                                 | contrib       |                      |                  | 
| 83  | <ul><li> - [ ] MXNetError    </li></ul>                            | base          |                      |                  | 
| 84  | <ul><li> - [ ] mx_float_p    </li></ul>                            | base          |                      |                  | 
| 85  | <ul><li> - [ ] py_str    </li></ul>                                | base          |                      |                  | 
| 86  | <ul><li> - [ ] print_summary    </li></ul>                         | visualization |                      |                  | 
| 87  | <ul><li> - [ ] Symbol.attr    </li></ul>                           | symbol        |                      |                  | 
| 88  | <ul><li> - [ ] Symbol.attr_dict    </li></ul>                      | symbol        |                      |                  | 
| 89  | <ul><li> - [ ] Symbol.debug_str    </li></ul>                      | symbol        |                      |                  | 
| 90  | <ul><li> - [ ] Symbol.grad    </li></ul>                           | symbol        |                      |                  | 
| 91  | <ul><li> - [ ] Symbol.list_attr    </li></ul>                      | symbol        |                      |                  | 
| 92  | <ul><li> - [ ] MXRecordIO.close    </li></ul>                      | recordio      |                      |                  | 
| 93  | <ul><li> - [ ] MXRecordIO.open    </li></ul>                       | recordio      |                      |                  | 
| 94  | <ul><li> - [ ] Optimizer.set_lr_mult    </li></ul>                 | optimizer     |                      |                  | 
| 95  | <ul><li> - [ ] Optimizer.set_lr_scale    </li></ul>                | optimizer     |                      |                  | 
| 96  | <ul><li> - [ ] Optimizer.set_wd_mult    </li></ul>                 | optimizer     |                      |                  | 
| 97  | <ul><li> - [ ] Optimizer.update    </li></ul>                      | optimizer     |                      |                  | 
| 98  | <ul><li> - [ ] LinearRegressionOutput    </li></ul>                | ndarray       | @naidu0830    |                  | 
| 99  | <ul><li> - [x] NDArray._at    </li></ul>                           | ndarray       |  @reminisce   |                  | 
| 100 | <ul><li> - [x] NDArray._slice    </li></ul>                        | ndarray       |   @reminisce  |                  | 
| 101 | <ul><li> - [x] NDArray._sync_copyfrom    </li></ul>                | ndarray       |  @reminisce    |                  | 
| 102 | <ul><li> - [ ] gamma    </li></ul>                                 | ndarray       |  @zhenwendai |                  | 
| 103 | <ul><li> - [ ] gammaln    </li></ul>                               | ndarray       |  @zhenwendai |                  | 
| 104 | <ul><li> - [ ] identity    </li></ul>                              | ndarray       | @zhenwendai |                  | 
| 105 | <ul><li> - [ ] load    </li></ul>                                  | ndarray       | @zhenwendai |                  | 
| 106 | <ul><li> - [ ] moveaxis    </li></ul>                              | ndarray       | @zhenwendai |                  | 
| 107 | <ul><li> - [ ] DataDesc    </li></ul>                              | io            |                      |                  | 
| 108 | <ul><li> - [ ] DataDesc.get_batch_axis    </li></ul>               | io            |                      |                  | 
| 109 | <ul><li> - [ ] DataDesc.get_list    </li></ul>                     | io            |                      |                  | 
| 110 | <ul><li> - [ ] DataIter.getpad    </li></ul>                       | io            |                      |                  | 
| 111 | <ul><li> - [ ] MXDataIter    </li></ul>                            | io            |                      |                  | 
| 112 | <ul><li> - [ ] ImageDetRecordIter    </li></ul>                    | io            |                      |                  | 
| 113 | <ul><li> - [ ] Initializer._legacy_init    </li></ul>              | initializer   |                      |                  | 
| 114 | <ul><li> - [ ] Initializer.dumps    </li></ul>                     | initializer   |                      |                  | 
| 115 | <ul><li> - [ ] ProgressBar    </li></ul>                           | callback      |   @saurabh3949                  |                  | 
| 116 | <ul><li> - [ ] log_train_metric    </li></ul>                      | callback      |                      |                  | 
| 117 | <ul><li> - [ ] module_checkpoint    </li></ul>                     | callback      |                      |                  | 
| 118 | <ul><li> - [ ] _parse_aux_states    </li></ul>                     | test_utils    |                      |                  | 
| 119 | <ul><li> - [ ] _parse_location    </li></ul>                       | test_utils    |                      |                  | 
| 120 | <ul><li> - [x] check_symbolic_backward    </li></ul>               | test_utils    |  @reminisce  |                  | 
| 121 | <ul><li> - [x] check_symbolic_forward    </li></ul>                | test_utils    | @reminisce  |                  | 
| 122 | <ul><li> - [ ] np_reduce    </li></ul>                             | test_utils    |                      |                  | 
| 123 | <ul><li> - [ ] numeric_grad    </li></ul>                          | test_utils    |                      |                  | 
| 124 | <ul><li> - [ ] print_max_err_loc    </li></ul>                     | test_utils    |                      |                  | 
| 125 | <ul><li> - [ ] BucketSentenceIter    </li></ul>                    | rnn           |                      |                  | 
| 126 | <ul><li> - [ ] encode_sentences    </li></ul>                      | rnn           |                      |                  | 
| 127 | <ul><li> - [ ] do_rnn_checkpoint    </li></ul>                     | rnn           |                      |                  | 
| 128 | <ul><li> - [ ] load_rnn_checkpoint    </li></ul>                   | rnn           |                      |                  | 
| 129 | <ul><li> - [ ] save_rnn_checkpoint    </li></ul>                   | rnn           |                      |                  | 
| 130 | <ul><li> - [ ] BaseRNNCell.begin_state    </li></ul>               | rnn           |                      |                  | 
| 131 | <ul><li> - [ ] BaseRNNCell.pack_weights    </li></ul>              | rnn           |                      |                  | 
| 132 | <ul><li> - [ ] BaseRNNCell.unpack_weights    </li></ul>            | rnn           |                      |                  | 
| 133 | <ul><li> - [ ] BaseRNNCell.unroll    </li></ul>                    | rnn           |                      |                  | 
| 134 | <ul><li> - [ ] BidirectionalCell    </li></ul>                     | rnn           |                      |                  | 
| 135 | <ul><li> - [ ] DropoutCell    </li></ul>                           | rnn           |                      |                  | 
| 136 | <ul><li> - [ ] FusedRNNCell    </li></ul>                          | rnn           |                      |                  | 
| 137 | <ul><li> - [ ] GRUCell    </li></ul>                               | rnn           |                      |                  | 
| 138 | <ul><li> - [ ] LSTMCell    </li></ul>                              | rnn           |                      |                  | 
| 139 | <ul><li> - [ ] ModifierCell    </li></ul>                          | rnn           |                      |                  | 
| 140 | <ul><li> - [ ] RNNCell    </li></ul>                               | rnn           |                      |                  | 
| 141 | <ul><li> - [ ] RNNParams    </li></ul>                             | rnn           |                      |                  | 
| 142 | <ul><li> - [ ] RNNParams.get    </li></ul>                         | rnn           |                      |                  | 
| 143 | <ul><li> - [ ] SequentialRNNCell    </li></ul>                     | rnn           |                      |                  | 
| 144 | <ul><li> - [ ] SequentialRNNCell.add    </li></ul>                 | rnn           |                      |                  | 
| 145 | <ul><li> - [ ] ZoneoutCell    </li></ul>                           | rnn           |                      |                  | 
| 146 | <ul><li> - [ ] BaseRNNCell._get_activation    </li></ul>           | rnn           |                      |                  | 
| 147 | <ul><li> - [ ] FusedRNNCell._slice_weights    </li></ul>           | rnn           |                      |                  | 
| 148 | <ul><li> - [ ] dump_profile    </li></ul>                          | profiler      |                      |                  | 
| 149 | <ul><li> - [ ] profiler_set_config    </li></ul>                   | profiler      |                      |                  | 
| 150 | <ul><li> - [ ] profiler_set_state    </li></ul>                    | profiler      |                      |                  | 
| 151 | <ul><li> - [ ] CustomOp    </li></ul>                              | operator      |                      |                  | 
| 152 | <ul><li> - [ ] CustomOp.backward    </li></ul>                     | operator      |                      |                  | 
| 153 | <ul><li> - [ ] CustomOp.forward    </li></ul>                      | operator      |                      |                  | 
| 154 | <ul><li> - [ ] CustomOpProp    </li></ul>                          | operator      |                      |                  | 
| 155 | <ul><li> - [ ] CustomOpProp.infer_shape    </li></ul>              | operator      |                      |                  | 
| 156 | <ul><li> - [ ] CustomOpProp.list_arguments    </li></ul>           | operator      |                      |                  | 
| 157 | <ul><li> - [ ] CustomOpProp.list_auxiliary_states    </li></ul>    | operator      |                      |                  | 
| 158 | <ul><li> - [ ] CustomOpProp.list_outputs    </li></ul>             | operator      |                      |                  | 
| 159 | <ul><li> - [x] ~~NDArrayOp.declare_backward_~~    </li></ul> | ~~operator~~  |                      |                  | 
| 160 | <ul><li> - [x] ~~NumpyOp~~    </li></ul>                           | ~~operator~~  |                      |                  | 
| 161 | <ul><li> - [x] ~~PythonOp~~    </li></ul>                          | ~~operator~~    |                      |                  | 
| 162 | <ul><li> - [x] ~~PythonOp.backward~~    </li></ul>        |~~operator~~      |                      |                  | 
| 163 | <ul><li> - [x] ~~PythonOp.forward~~    </li></ul>                  | ~~operator~~   |                      |                  | 
| 164 | <ul><li> - [x] ~~PythonOp.get_symbol~~    </li></ul>               | ~~operator~~      |                      |                  | 
| 165 | <ul><li> - [x] ~~PythonOp.infer_shape~~    </li></ul>                |~~operator~~  |                      |                  | 
| 166 | <ul><li> - [x] ~~PythonOp.list_arguments~~    </li></ul>           | ~~operator~~     |                      |                  | 
| 167 | <ul><li> - [x] ~~PythonOp.list_outputs~~    </li></ul>                 | ~~operator~~      |                  |                  | 
| 168 | <ul><li> - [x] ~~PythonOp.need_top_grad~~    </li></ul>                | ~~operator~~     |                  |                  | 
| 169 | <ul><li> - [ ] register    </li></ul>                              | operator      |                      |                  | 
| 170 | <ul><li> - [ ] CustomOpProp.infer_type    </li></ul>               | operator      |                      |                  | 
| 171 | <ul><li> - [ ] NameManager.get    </li></ul>                       | name          |                      |                  | 
| 172 | <ul><li> - [ ] Prefix    </li></ul>                                | name          |                      |                  | 
| 173 | <ul><li> - [ ] Monitor.install    </li></ul>                       | monitor       |                      |                  | 
| 174 | <ul><li> - [ ] Monitor.tic    </li></ul>                           | monitor       |                      |                  | 
| 175 | <ul><li> - [ ] Monitor.toc    </li></ul>                           | monitor       |                      |                  | 
| 176 | <ul><li> - [ ] Monitor.toc_print    </li></ul>                     | monitor       |                      |                  | 
| 177 | <ul><li> - [ ] getLogger    </li></ul>                             | log           |                      |                  | 
| 178 | <ul><li> - [x] KVStore.load_optimizer_states    </li></ul>         | kvstore       | @eric-haibin-lin  |                  | 
| 179 | <ul><li> - [x] KVStore.save_optimizer_states    </li></ul>         | kvstore       |  @eric-haibin-lin  |                  | 
| 180 | <ul><li> - [x ] KVStore.set_optimizer    </li></ul>                 | kvstore       |  @eric-haibin-lin   |                  | 
| 181 | <ul><li> - [x] create    </li></ul>                                | kvstore       | @eric-haibin-lin  |                  | 
| 182 | <ul><li> - [ ] ImageIter    </li></ul>                             | image         |                      |                  | 
| 183 | <ul><li> - [ ] ImageIter.next    </li></ul>                        | image         |                      |                  | 
| 184 | <ul><li> - [ ] ImageIter.next_sample    </li></ul>                 | image         |                      |                  | 
| 185 | <ul><li> - [ ] ImageIter.reset    </li></ul>                       | image         |                      |                  | 
| 186 | <ul><li> - [x] Executor    </li></ul>                              | executor      |   @kevinthesun |                  | 
| 187 | <ul><li> - [x] Executor.backward    </li></ul>                     | executor      |   @kevinthesun |                  | 
| 188 | <ul><li> - [x] Executor.copy_params_from    </li></ul>             | executor      |  @kevinthesun |                  | 
| 189 | <ul><li> - [x] Executor.reshape    </li></ul>                      | executor      | @kevinthesun  |                  | 
| 190 | <ul><li> - [x] Executor.set_monitor_callback    </li></ul>         | executor      |  @kevinthesun  |                  | 
| 191 | <ul><li> - [ ] Executor._get_outputs    </li></ul>                 | executor      |                      |                  | 
| 192 | <ul><li> - [x] Executor.debug_str    </li></ul>                    | executor      | @kevinthesun  |                  | 
| 193 | <ul><li> - [ ] compute_gradient    </li></ul>                      | contrib       |                      |                  | 
| 194 | <ul><li> - [ ] grad_and_loss    </li></ul>                         | contrib       |                      |                  | 
| 195 | <ul><li> - [ ] mark_variables    </li></ul>                        | contrib       |                      |                  | 
| 196 | <ul><li> - [ ] set_recording    </li></ul>                         | contrib       |                      |                  | 
| 197 | <ul><li> - [ ] LogMetricsCallback    </li></ul>                    | contrib       |                      |                  | 
| 198 | <ul><li> - [ ] TrainingStateScope    </li></ul>                    | contrib       |                      |                  | 
| 199 | <ul><li> - [ ] build_param_doc    </li></ul>                       | base          |                      |                  | 
| 200 | <ul><li> - [ ] c_array    </li></ul>                               | base          |  @yuruofeifei   |                  | 
| 201 | <ul><li> - [ ] c_str    </li></ul>                                 | base          |  @yuruofeifei |                  | 
| 202 | <ul><li> - [ ] check_call    </li></ul>                            | base          |                      |                  | 
| 203 | <ul><li> - [ ] ctypes2buffer    </li></ul>                         | base          |                      |                  | 
| 204 | <ul><li> - [ ] ctypes2numpy_shared    </li></ul>                   | base          |                      |                  | 
| 205 | <ul><li> - [ ] _load_lib    </li></ul>                             | base          |                      |                  | 
| 206 | <ul><li> - [ ] _notify_shutdown    </li></ul>                      | base          |                      |                  | 
| 207 | <ul><li> - [ ] add_fileline_to_docstring    </li></ul>             | base          |                      |                  | 
| 208 | <ul><li> - [ ] AttrScope    </li></ul>                             | attribute     |                      |                  | 
| 209 | <ul><li> - [ ] AttrScope.current    </li></ul>                     | attribute     |                      |                  | 
| 210 | <ul><li> - [ ] AttrScope.get    </li></ul>                         | attribute     |                      |                  |

**Comment by piiswrong** [comment]
NumpyOP pythonop and ndarrayop are deprecated. We only support customop now

**Comment by naidu0830** [comment]
I will document "LinearRegressionOutput" API

**Comment by asmushetzel** [comment]
I will document all samplers and randoms as I anyway enhanced or authored them, namely 

 random_gamma 
random_exponential
random_poisson
random_negative_binomial 
random_generalized_negative_binomial

sample_uniform
sample_normal
sample_gamma 
sample_exponential
sample_poisson
sample_negative_binomial 
sample_generalized_negative_binomial

I may need more than 2 business days but should be definitely done by end of next week.

**Comment by moontails** [comment]
I will document Functions from 125 to 147 in the rnn package.

If I get more time, I will also work on 161-168

**Comment by domdivakaruni** [comment]
@moontails that is tremendous thank you!  Do you plan on making PR's or creating Issues with your functions in markdown? When you are done with your first function send it in so we can help with a review before you get rolling with the rest. great stuff!

**Comment by moontails** [comment]
@domdivakaruni I was thinking of creating PR's. 

Thats a good idea! I will submit a PR when am done with the changes for the first function and gather all feedback before diving in on the rest :)

**Comment by zhenwendai** [comment]
I can document some of ndarray operators: 

102 gamma ndarray 		
103 gammaln ndarray 		
104 identity ndarray 		
105 load ndarray 		
106 moveaxis ndarray

**Comment by domdivakaruni** [comment]
Hi @moontails do you want to call ownership of all 22 rnn functions at once or do you want to stake your claim in batches as you make progress?

**Comment by yuruofeifei** [comment]
I will work on zeros_like, ones_like, softmax_cross_entropy, c_array, c_str

**Comment by jhwei** [comment]
I can work on resize_short, scale_down.

**Comment by domdivakaruni** [comment]
@yuruofeifei @jhwei @zhenwendai good stuff!

**Comment by moontails** [comment]
@domdivakaruni either works fine for me. I had hoped to send out a PR yesterday to get some feedback before working on the rest, but ran into some issues setting up the repo (with the whole installing mxnet and make lint)

**Comment by domdivakaruni** [comment]
@moontails did you get it worked out? Let me know what you ran into.
@nswamy fyi

**Comment by nswamy** [comment]
@moontails whats the issue that you are running into?
@domdivakaruni can we have folks picks few APIs at once and add more to their bucker as they make progress?

**Comment by Lyken17** [comment]
I can document 6~11.

Question : Should we use `reST` or `Markdown` for improvement through github issues? This page says `Markdown`, but [pull request pages](https://s3.amazonaws.com/mxnetdoc/API+Doc+Contributions%281%29.pdf) requires `reST`.

**Comment by nswamy** [comment]
@Lyken17 please prefer to use `reST` and submit a PR. If you want to just submit doc outside of the  the code(python doc string or describe method of C++ code)  use either `reST` or  `Markdown` and submit an issue.

**Comment by nswamy** [comment]
@zhenwendai, The docs for these APIs are already updated. Could you please pick a different set?
102 gamma ndarray
103 gammaln ndarray
104 identity ndarray

**Comment by srviest** [comment]
I will work on 58 color_normalize. BTW, where are the definitions of 78 fft & 79 ifft?

**Comment by jiayue666** [comment]
I can document 2-5.

**Comment by szha** [comment]
I will take 130-145.

@piiswrong do we need to document 146 BaseRNNCell._get_activation and 147 FusedRNNCell._slice_weights? These are not intended for use other than by the APIs.

**Comment by domdivakaruni** [comment]
@moontails are you still working on these APIs? What can you share with @szha? can you please let us know as soon as you can? Thanks!

**Comment by moontails** [comment]
@domdivakaruni @szha I am sorry guys, I still havent been able to get it to work on my Macbook. 

I will let @szha start checking things off from 130-145 and get back on track if I am able to unblock myself.

**Comment by asmushetzel** [comment]
@moontails Don't know which problem you ran into but also had to resolve some hiccups on my mac. Resolved it by:
- Complete fresh clone of the whole mxnet
- Not running all build/installs for this repo. Just the documentation creation as described in the doc referenced in section 4. at the beginning of this thread. 
- In file tests/ci_build/install/ubuntu_install_python.sh bypassed the certificate checking by 
       wget https://bootstrap.pypa.io/get-pip.py --no-check-certificate

That got me going. I used a separate mxnet repo for standard building/installation and functional testing (I had to touch a bit more than just the "describe" section of the operators). Don't know whether this was necessary, but at least it worked that way. 

Did punt on lint on my mac as I couldn't get it to work. Rather did submit the pull request and fixed the flagged lint issues based on the automatic tests. It's straightforward.

**Comment by eric-haibin-lin** [comment]
I'm going to take the four APIs for kvstore 178 - 181

**Comment by nswamy** [comment]
@Lyken17 any luck updating the doc for the APIs that you picked ?

**Comment by Lyken17** [comment]
Hi @nswamy 

I just updated GridGenerator and submit it to issue https://github.com/dmlc/mxnet/issues/6147.

I meet troubles when documenting `ndarray.UpSampling`. Even myself cannot find the proper way to use it, according to issues https://github.com/dmlc/mxnet/issues/2823 https://github.com/dmlc/mxnet/issues/4134 https://github.com/dmlc/mxnet/issues/1412, it seems to be an incomplete implementation. I also looked up several FCN mxnet implementations, they all use `deconvolution` instead of `UpSampling`.

**Comment by nswamy** [comment]
@Lyken17, Thanks for posting the issue, I will convert this to actual documentation in code. 
You can skip UpSampling for now, we might need to add a note that it is incomplete. @mli, @piiswrong what do you suggest?

For other APIs that you work on, can you please edit the docs in the code directly? If building and rendering takes time or you get into an issue, you can skip that step and just follow the tips to avoid any lint errors.

**Comment by apaleyes** [comment]
Hi @nswamy 

I've just posted a pr for 118 and 119: https://github.com/dmlc/mxnet/pull/6314

One thing though is that I wasn't able to get rendering going on my Mac. "make lint" was fine, but building docs was always failing. I'll post a separate issue perhaps to get some advice on that.
